python/src/trezorlib/cardano.py

```diff
@@ -315,6 +321,74 @@ def create_withdrawal(withdrawal) -> messages.CardanoTxWithdrawalType:
     )
 
 
+def create_auxiliary_data(auxiliary_data) -> messages.CardanoTxAuxiliaryDataType:
+    if auxiliary_data is None:
+        return None
+
+    AUXILIARY_DATA_MISSING_FIELDS_ERROR = ValueError(
+        "Auxiliary data is missing some fields"
+    )
+
+    if "type" not in auxiliary_data:
+        raise AUXILIARY_DATA_MISSING_FIELDS_ERROR
+
+    auxiliary_data_type = auxiliary_data["type"]
+    if auxiliary_data_type == messages.CardanoAuxiliaryDataType.BLOB:
+        if "blob" not in auxiliary_data:
+            raise AUXILIARY_DATA_MISSING_FIELDS_ERROR
+
+        return messages.CardanoTxAuxiliaryDataType(
+            type=auxiliary_data_type,
+            blob=bytes.fromhex(auxiliary_data["blob"]),
+        )
+    elif auxiliary_data_type == messages.CardanoAuxiliaryDataType.TUPLE:
+        # TODO CATA: should this be checked? I can't really even test it on FW level since it's
```

Comment:
I'd rather not include a check for the non-presenc of other fields - that doesn't really scale and the caller should notice they are passing something weird if they pass both the blob and metadata field

Reply:
https://github.com/vacuumlabs/trezor-firmware/pull/57/commits/39dddb339a5c4ace8e56f62818ee8a2330a123a2